### ISSUE TITLE
 The Radio from react-aria-components is missing excludeFromTabOrder prop 

### ISSUE THREAD

**Issue by KelvinOm** [adobe/react-spectrum#6569]
### Provide a general summary of the issue here

 The Radio from react-aria-components is missing excludeFromTabOrder prop 

### 🤔 Expected Behavior?

There is possibility to exclude the Radio component from the sequential tab order 

### 😯 Current Behavior

There is no way to exclude the Radio component from the sequential tab order 

### 🖥️ Steps to Reproduce

Simply try to exclude the Radio from tab order.

### Version

react-aria-components: 1.2.1

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

macOS

**Comment by snowystinger** [comment]
Can you share an example for why you need to exclude a radio from the tab order?

**Comment by akyllus** [comment]
<html>
<body>
<p style="box-sizing: border-box; margin: 0px 0px 1rem; padding: 0px; color: rgb(255, 255, 255); font-family: &quot;Dancing Script&quot;, cursive; font-size: 14.4px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">To address the issue of the missing<span> </span><code style="box-sizing: border-box; margin: 0px; padding: 0px; font-family: SFMono-Regular, Menlo, Monaco, Consolas, &quot;Liberation Mono&quot;, &quot;Courier New&quot;, monospace; font-size: 12.6px; color: rgb(232, 62, 140); overflow-wrap: break-word;">excludeFromTabOrder</code><span> </span>prop in the<span> </span><code style="box-sizing: border-box; margin: 0px; padding: 0px; font-family: SFMono-Regular, Menlo, Monaco, Consolas, &quot;Liberation Mono&quot;, &quot;Courier New&quot;, monospace; font-size: 12.6px; color: rgb(232, 62, 140); overflow-wrap: break-word;">Radio</code><span> </span>component from<span> </span><code style="box-sizing: border-box; margin: 0px; padding: 0px; font-family: SFMono-Regular, Menlo, Monaco, Consolas, &quot;Liberation Mono&quot;, &quot;Courier New&quot;, monospace; font-size: 12.6px; color: rgb(232, 62, 140); overflow-wrap: break-word;">react-aria-components</code>, we need to modify the component to accept this prop and then use it to control the tab order behavior. Below is an example of how this could be implemented:</p><ol style="box-sizing: border-box; margin: 0px; padding: 0px; color: rgb(255, 255, 255); font-family: &quot;Dancing Script&quot;, cursive; font-size: 14.4px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li style="box-sizing: border-box; margin: 0px 0px 0.7rem; padding: 0px; display: flex; flex-direction: column; justify-content: flex-start; align-items: flex-start;"><strong style="box-sizing: border-box; margin: 0px; padding: 0px; font-weight: bolder;">Update the Radio Component:</strong></li></ol><pre class="code-badge-pre" style="box-sizing: border-box; margin: 0px 0px 1rem; padding: 0px; font-family: SFMono-Regular, Menlo, Monaco, Consolas, &quot;Liberation Mono&quot;, &quot;Courier New&quot;, monospace; font-size: 12.6px; overflow: auto; display: block; color: rgb(33, 37, 41); position: relative; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="code-badge" style="box-sizing: border-box; margin: 0px; padding: 5px 8px; display: flex; flex-direction: row; white-space: normal; background: rgb(51, 51, 51); color: white; font-size: 0.875em; opacity: 0.5; transition: opacity 0.5s linear 0s; border-radius: 0px 0px 0px 7px; position: absolute; right: 0px; top: 0px;"><div class="code-badge-language" style="box-sizing: border-box; margin: 0px 10px 0px 0px; padding: 0px; font-weight: 600; color: goldenrod;">jsx</div><div title="Copy to clipboard" style="box-sizing: border-box; margin: 0px; padding: 0px;"><i class="fa fa-copy code-badge-copy-icon" style="box-sizing: border-box; margin: 0px; padding: 0px 7px; display: inline-block; font: 1.2em / 1 FontAwesome; text-rendering: auto; -webkit-font-smoothing: antialiased; cursor: pointer;"></i></div></div><code class="language-jsx hljs language-javascript" data-highlighted="yes" style="box-sizing: border-box; margin: 0px; padding: 1em; font-family: SFMono-Regular, Menlo, Monaco, Consolas, &quot;Liberation Mono&quot;, &quot;Courier New&quot;, monospace; font-size: inherit; color: rgb(221, 221, 221); overflow-wrap: break-word; word-break: normal; background: rgb(48, 48, 48); display: block; overflow-x: auto;">
  | import React, { useRef } from 'react';
-- | --
  | import { useFocusable } from '@react-aria/focus';
  |  
  | function Radio({ children, isDisabled, excludeFromTabOrder, ...props }) {
  | const ref = useRef(null);
  | const { focusableProps } = useFocusable(
  | {
  | ...props,
  | isDisabled,
  | excludeFromTabOrder,
  | },
  | ref
  | );
  |  
  | return (
  | <label {...focusableProps} ref={ref}>
  | <input type="radio" disabled={isDisabled} />
  | {children}
  | </label>
  | );
  | }
  |  
  | export default Radio;

</code></pre>
</body>
</html>

**Comment by KelvinOm** [comment]
> Can you share an example for why you need to exclude a radio from the tab order?

In our app, we use components only for presentation in some mode, without the ability to interact with them. Regardless, I believe that the behavior of these components should be consistent throughout the library(other components have `excludeFromTabOrder` prop).

**Comment by snowystinger** [comment]
I don't think it makes sense for Radio to be excluded from tab order.
Radios only exist in a RadioGroup and RadioGroup is itself a single tab stop, arrow keys to move around inside it.

You could use the RadioGroup as readonly. Then the information is still available to all users, not just the sighted users.

If it's purely visual and is just decoration, then you don't need to use a RadioGroup or Radio at all, just style some divs to look like one.

I'll double check if that prop should be exposed on so many things. It seems further reaching than I had remembered. Adding it to RadioGroup would be the thing to do if we're making it match the others though.

**Comment by KelvinOm** [comment]
@snowystinger Yes, you are absolutely right. It is best to add this prop to the `RadioGroup`.

**Comment by KelvinOm** [comment]
@snowystinger I also noticed that it is also not possible to specify the prop for the Link component. I also can't pass the tabIndex prop, which is a valid parameter for the link.

**Comment by snowystinger** [comment]
Chatted with the team, adding to RadioGroup is the right move.
Given Button has it, Link should be fine as well.
Thanks for bringing it up.

Want to open a PR with the change?

**Comment by KelvinOm** [comment]
@snowystinger I'm willing to do this and will try to find some time this week.

**Comment by LFDanLu** [comment]
@KelvinOm Hi, sorry about the back and forth, but the team revisited this recently and have pivoted a bit on our decision here. It feels like exposing `excludeFromTabOrder` isn't actually the correct approach here because setting that prop alone doesn't actually accomplish the exact behavior you'd want here. Since you mentioned that you have a presentational RadioGroup/Radio in your app that isn't actually interactive (aka it is there to just display a selected option I'm guessing), you'd actually would want to make that RadioGroup/Radio disabled/inert instead. Setting `excludeFromTabOrder` would just make it not keyboard tabbable but clicking a radio would still move focus to it which it seems you wouldn't want either. Let us know if that makes sense or if we've misunderstood the exact use case here.

**Comment by KelvinOm** [comment]
@LFDanLu Hi, I think it's not quite right to make the decision based only on my example. So, if we abstract from it, then the ability to exclude a component from tab navigation should still be present. Because it can be useful to have a such ability. 

[Here](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex), for example, is what MDN writes about it.

> Note: tabindex="-1" may be useful for elements that should not be navigated to directly using the Tab key, but need to have keyboard focus set to them. Examples include an off-screen modal window that should be focused when it comes into view, or a form submission error message that should be immediately focused when an errant form is submitted.

I also believe that the behavior of interactive components in the library should be consistent. Why do checkbox or button have same attribute? 

I hope these arguments make sense to you.

**Comment by LFDanLu** [comment]
I agree that there are cases that may require a element to be excluded from tab navigation but still require the ability to programmatically focus them, like for instance the button in a combobox or the clear button in a search field. The excludeFromTabOrder prop was originally introduced for those cases but it seems it was exposed on several other components because those components used to expose a `tabIndex` prop due to legacy reasons/on accident. IMO its a bit regrettable that it was exposed on such a wide set of components via the FocusableDOMProps so either we make it consistent by adding it to ALL focusable elements (runs the risk that users overwrite a tabindex that we are providing/managing) or we hold off until in favor of people disabling/making an element inert when appropriate. I'd be fine adding it to all components but just worried it be prone to misuse when there are viable/more appropriate behavior alternatives

**Comment by KelvinOm** [comment]
I fully share your concerns, but sometimes there is still a need to use properties that are better not to use. I treat `excludeFromTabOrder` prop the same way as I do `UNSAFE_className ` or `UNSAFE_style`. I think developers should still have a way to manage `tabIndex`, but this should be a conscious decision, since in general it is better not to use it.

**Comment by snowystinger** [comment]
During the most recent discussion on this topic, we reviewed reasons people had asked for this prop. So far, each case has been solved without exposing this, and usually, it would have led to worse accessibility. The team is of the opinion that we accidentally exposed it on a few too many components and we'd prefer not to expose it on more at this point in time. I'm going to close this issue now as consistency alone is not enough of a reason. If another use case comes up, we can revisit. Thank you for all of your input and for your patience and understanding.

Typical cases have been solved by using readonly or disabled on the component. Or wrapping the component in an `inert` element.

**Comment by KelvinOm** [comment]
@snowystinger @LFDanLu The final question: Are you considering removing those props that were accidentally exposed? I would like to understand the scope of the changes to our codebase in order to better plan for future updates.

**Comment by snowystinger** [comment]
As of this point, we are not considering removing those. We realize that people may already be using them, and this would be breaking.
